### PR TITLE
{tools}[foss/2018a,intel/2018a] ToFu 1.4.0 w/ Python 3.6.4

### PR DESCRIPTION
**Pull request by Louwrensth**

### CODE REVIEW

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- tofu-1.4.0/setup.py.orig	2019-05-29 00:08:47.000000000 +0200
++++ tofu-1.4.0/setup.py	2019-05-29 00:08:42.000000000 +0200
+@@ -64,8 +64,8 @@
+     os.environ['CC'] = 'gcc-8'
+     os.environ['CXX'] = 'gcc-8'
+ else:
+-    os.environ['CC'] = 'gcc'
+-    os.environ['CXX'] = 'gcc'
++    os.environ['CC'] = 'icc'
++    os.environ['CXX'] = 'icpc'
```
Comment (akesandgren) [inline easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel.patch]: A better patch for EasyBuild is (probably) to let it pick up the already defined env vars for CC/CXX, ie to drop everything that tries to define os.environ of CC/CXX in setup.py

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel-2018a-Python-3.6.4.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+description = """Tomography for Fusion."""
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+
+dependencies = [
+    ('Python', '3.6.4'),
+    ('matplotlib', '2.1.2', versionsuffix),
+]
+
+exts_default_options = {'source_urls': [PYPI_LOWER_SOURCE]}
```
Comment (akesandgren) [inline easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel-2018a-Python-3.6.4.eb]: use_pip must be explicitly set in PythonBundle based easyconfigs, and should normally be set to True.

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel-2018a-Python-3.6.4.eb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            '6aae61d5eef6b7389fdae3eaac7bc9d6e045a1e75f897209516c4f2605f14855',  # ToFu-1.4.0-no-git.patch
+            '96013dc65891eb9645ff7efed1190846b41cd837ebec0827436a8673ac4589c1',  # ToFu-1.4.0-intel.patch
+        ],
+    }),
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
```
Comment (akesandgren) [inline easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel-2018a-Python-3.6.4.eb]: This sanity_check_paths is the default for PythonBundle and can be removed

Comment (Louwrensth) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-intel-2018a-Python-3.6.4.eb]: Thanks a lot for your hints. I've incorporated them all and squashed the commit history. Hope it builds :)

**Comment by Louwrensth** [comment]
It seems to pass well now. Thanks again, @akesandgren for the hints.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 23 out of 31 (4 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/2b3d16bdaebe36b1ae74e59397b30787 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/11ed20040d0ada26a565b3c29677f142 for a full test report.

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch [hunk 4]
```diff
@@ -0,0 +1,19 @@
+diff --git a/setup.py b/setup.py
```
Comment (akesandgren) [inline easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: Please add description and author info to the patches.
And update checksums accordingly.

Comment (verdurin) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: @Louwrensth ping - any update on this?

**Comment by Louwrensth** [comment]
Thanks for the bump. Will do.

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch (hunk 4 above)
Comment (Louwrensth) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: Thanks for the bump. Will do.

Comment (Louwrensth) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: Done.

Comment (Louwrensth) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: @verdurin, @akesandgren : should I do anything else? the PR still says it's waiting for some request...

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/77fdced6c5e5c6d5c6c3122305edd704 for a full test report.

File: easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch (hunk 4 above)
Comment (akesandgren) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: Doing a force push destroys the history and confuses travis, that's likely why it thinks there are unresolved review requests.

And, unless I'm blind, ToFu-1.4.0-foss-2018a-Python-2.7.14.eb is missing checksums for one patch and so is ToFu-1.4.0-intel-2018a-Python-2.7.14.eb

And the checksum for ToFu-1.4.0-no-git.patch is wrong for all easyconfigs.

Comment (akesandgren) [reply easybuild/easyconfigs/t/ToFu/ToFu-1.4.0-CC-by-env.patch]: I recommend eb *.eb --inject-checksum --force

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/07718a6e01bda9847e8359e2d0dfcaff for a full test report.

**Comment by akesandgren** [comment]
Going in, thanks @Louwrensth!

**Pull request merged**